Docs: Add how to replace home dashboard (#26237)
grafana/grafana: docs/sources/administration/home_dashboard.md

> +++

Reviewer: Please change the file name to change-home-dashboard.md. Always match the topic title as closely as possible.

> title = "How to replace the default home dashboard"

Reviewer: ```suggestion
title = "Change the default home dashboard"
```

description = "How to replace the default home dashboard"
keywords = ["grafana", "configuration", "documentation", "home"]
type = "docs"
[menu.docs]
name = "Home Dashboard"
identifier = "home-dashboard"
parent = "admin"
weight = 100
+++

> # How to replace the default home dashboard

Reviewer: ```suggestion
# Change the default home dashboard
```

> You can easily replace the default homepage of Grafana in two different ways.

Reviewer: ```suggestion
You can change the default homepage of Grafana in two different ways: change Grafana preferences or change the Grafana configuration.
```

Reviewer: Avoid saying something is "easy" or "simple" when giving directions. If someone has trouble, they will feel bad.

Reviewer: Also, will this affect other users or just you?

Author: Each section has its own explanation of who will be impacted. Do you think it's enough or should I add them here?

Reviewer: When I got to this sentence, the question I had was "is this change for me or will it affect all users of this instance?" If I wonder this, others will. I suggest stating that some methods change it for the individual user, other methods change it for all users of the instance, and that you note in each section what the scope of the change is.

Reviewer 2: Reading the docs I think the information might be better structured around if you want to change the default dashboard for the organsation or for the user. I think you usually come searching for information on either rather than both. I feel the shared instruction for how to change a dashboard through preferences became a bit harder to understand than strictly necessary.

This is my attempt at a restructuring. What do you think?

---

You can change the default dashboard both on the organsation and user level. The dashboard you set as your default for the whole organisation is the one all users will see by default. Individual users can set their own default home dashboard and it will overrride whatever is set on the organisation level.

## Change Organsation default dashboard

### Setting default dashboard through preferences

### Setting default dashboard through configuration

## Change user default dashboard

only prefernces apply here

Author: Yes it's a good idea! I did a v3 according to this structure, I prefer this version. What do you think?

> ## Preferences

Reviewer: ```suggestion
## Change home dashboard preferences
```

> You can choose any of your existing Grafana dashboards to replace the default home dashboard in the preferences. First, you need to star the dashboard that you want to use. Then, in `Configuration > Preferences`, you can select one of your starred dashboards in the `Home Dashboard` input. It will set it for all the users in your organization.

Reviewer: ```suggestion
You can choose any of your existing Grafana dashboards to replace the default home dashboard in the preferences. 

1. Star the dashboard that you want to use. 
1. In **Configuration > Preferences**, select one of your starred dashboards in the **Home Dashboard** field. 

Changing the preference changes the home dashboard for all the users in your organization.
```

Reviewer: These instructions are incomplete. WHERE do you star the dashboard? How do you do it?

Also, you need to make clear before the instructions who this change will affect and how to affect just you vs. the whole organization.

You can also replace it for you only by updating your own preferences.

> If you change the home dashboard in both preferences, user preferences will override organization preferences. 

Reviewer: The first part of this sentence makes no sense to me. 

Move the bit about user preferences overriding organization preferences before the instructions.

> ## Configuration

Reviewer: ```suggestion
## Change home dashboard configuration
```

> You can also provide your own JSON file and set the path to this file in the [configuration]({{< relref "./configuration.md">}}) file.

Reviewer: ```suggestion
You can provide your own JSON file and set the path to this file in the [configuration]({{< relref "./configuration.md">}}) file.
```

Reviewer: I recommend linking to the specific heading with the setting they need to update.

Reviewer: Also, be sure to mention that this should be in their custom.ini, not their default.ini.

Author: I think it's best to keep the generic link here because it's for people that don't know how to update their configuration file (and I don't want to repeat all the ways of doing so here). There isn't more information on the setting in the configuration documentation than in this file. Also for adding the mention to custom.ini, I'm not sure. It's grafana.ini on other OS and I think if the user doesn't know this, he should read the documentation on configuration first.

Maybe I should just rephrase it with something like: `Update your configuration file to set the path to the JSON file. Read how to update this file in the [configuration]({{< relref "./configuration.md">}}) documentation`?

Reviewer: Yes, I like that change. You're right, it will help address users of varying skill levels.

```ini
[dashboards]
# Path to the default home dashboard. If this value is empty, then Grafana uses StaticRootPath + "dashboards/home.json"

> default_home_dashboard_path =

Reviewer: In the custom.ini, there is usually a `;`. Users need to remove it before this command is valid.

Reviewer: Actual instructions, preferably in numbered step format, would be great here. An example of the link format and perhaps a link to instructions for getting dashboard JSON files would also be helpful.

```

> If preferences are set as described above, they will override this value.

Reviewer: ```suggestion
If preferences are set as described above, then they override this value.
```

Reviewer: Put this first, before the instructions.
